File: ContainsDuplicate/ContainsDuplicate.py
from typing import List


def hasDuplicate(nums: List[int]) -> bool:
        seen = set()
        for i in nums:
            if(i in seen):
                return True
            else:
                seen.add(i)
        return False


nums = [1, 2, 3, 4]
print(hasDuplicate(nums))

File: ContainsDuplicate/ContainsDuplicate.java
package ContainsDuplicate;

import java.util.*;
public class ContainsDuplicate {
    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        System.out.println(solution(nums));
    }

    public static boolean solution(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            } else {
                seen.add(num);
            }
        }
        return false;
    }
}
